Sprint_5_first commit с тестами и описанием локаторов

tomilinbvGit opened the pull request.

TomilinBV committed:
first commit с тестами и описанием локаторов

AlexKitIt commented on locators:
```diff
@@ -0,0 +1,43 @@
+Локаторы для страницы регистрации:
+
+# Локатор для "Имя"
+#root > div > main > div > form > fieldset:nth-child(1) > div > div > input
```
Нужно исправить: в этом модуле должны храниться локаторы записанные в переменные, для использования в проекте

AlexKitIt replied:
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от root и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

AlexKitIt commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,34 @@
+class TestConstructorSection:
+    #Тестируем скрол к секции булок (сначала скролим до начинки, потом возвращаемся)
+    def test_constructor_jump_to_buns_section(self):
+        driver = webdriver.Chrome()
```
Нужно исправить: здесь и далее: инстанс драйвера должен подниматься и закрываться в фикстуре

AlexKitIt commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,34 @@
+    #Тестируем скрол к секции булок (сначала скролим до начинки, потом возвращаемся)
+    def test_constructor_jump_to_buns_section(self):
+        driver = webdriver.Chrome()
+        driver.get("https://stellarburgers.nomoreparties.site")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, для них стоит создать модуль urls или отдельный класс в data

AlexKitIt commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,34 @@
+        # Находим поле "Email" и заполняем его данными заранее зарегистрированного пользователя
+        driver.find_element(By.CSS_SELECTOR,
+                            '#root > div > main > div > form > fieldset:nth-child(1) > div > div > input').send_keys(
+            '45697833@ya.ru')
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тестах и тестовых классах их держать не стоит

AlexKitIt commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,34 @@
+
+class TestConstructorSection:
+    #Тестируем скрол к секции булок (сначала скролим до начинки, потом возвращаемся)
+    def test_constructor_jump_to_buns_section(self):
```
Нужно исправить: тестов на переход по табам должно быть три отдельных, по одному на каждый таб

AlexKitIt commented on tests/test_constructor_section.py:
```diff
@@ -0,0 +1,34 @@
+
+        # Возвращаемся к булкам
+        element2 = driver.find_element(By.XPATH, '//*[@id="root"]/div/main/section[1]/div[2]/h2[1]')
+        driver.execute_script("arguments[0].scrollIntoView();", element2)
```
Нужно исправить: здесь и далее: тест всегда заканчивается assert где происходит сравнение ожидаемого результата с фактическим. Тут его нет, проверку необходимо дописать

AlexKitIt commented on tests/test_registration.py:
```diff
@@ -0,0 +1,57 @@
+        # Находим кнопку "Регистрация" и кликаем
+        driver.find_element(By.XPATH, '//*[@id="root"]/div/main/div/form/button').click()
+
+        driver.implicitly_wait(5)
```
Нужно исправить: здесь и далее: слипов и неявных ожиданий быть не должно. Необходимо использовать expected_conditions и встраивать ожидания на нем.

AlexKitIt commented on tests/test_registration.py:
```diff
@@ -0,0 +1,57 @@
+        driver.find_element(By.CSS_SELECTOR, '#root > div > main > div > form > fieldset:nth-child(1) > div > div > input').send_keys('Test User29494895811343433')
+
+        # Находим поле "Email" и заполняем его
+        driver.find_element(By.CSS_SELECTOR, '#root > div > main > div > form > fieldset:nth-child(2) > div > div > input').send_keys('test15699@ya.ru')
```
Нужно исправить: для теста регистрации email должен генерироваться каждый раз новый, в противном случае мы получим ошибку уже созданного пользователя. Необходимо написать метод, генерирующий email и положить его в helpers

TomilinBV committed:
Second commit (commit после перехода в 15 поток).
Локаторы переписаны и расположены в файле TestLocators, по каждому локатору создана описание.
Фикстура для запуская драйвера расположена в файле conftest.
Написаны генераторы почты и пароля (расположены в файле helpers).
В файле data расположены классы с тестовыми данными и используемы URL.
Файл README.md дополнен описанием проекта.

AlexKitIt commented on TestLocators.py:
Нужно исправить: именование пакетов, модулей, методов в питоне ведется в snake_case, нейминг необходимо исправить в соответствии с этим правилом

AlexKitIt commented on locators:
Нужно исправить: этот модуль получается нам не нужен теперь
